modules/persistent-storage-csi-smb-cifs-provision-dynamic.adoc

```diff
@@ -0,0 +1,130 @@
+// Module included in the following assemblies:
+//
+// * storage/container_storage_interface/persistent-storage-csi-smb-cifs.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="persistent-storage-csi-smb-cifs-provision-dynamic_{context}"]
+= Dynamic provisioning
+
+You can create a storage class for dynamic provisioning of Common Internet File System (CIFS) dialect/Server Message Block (SMB) protocol volumes.
+
+.Prerequisites
+* {FeatureName} CSI Driver Operator and driver installed.
+* You are logged in to the running {product-title} cluster.
+* You have installed the SMB server and know the following information about the server:
+** Hostname
+** Share name
+** Username and password
+
+.Procedure
+
+To set up dynamic provisioning:
+
+. Create a Secret for access to the Samba server using the following command with the following example YAML file:
++
+[source,terminal]
+--
+$ oc create -f <file_name>.yaml
+--
++
+[source,yaml]
+.Secret example YAML file
+--
+apiVersion: v1
+kind: Secret
+metadata:
+  name: smbcreds <1>
+  namespace: samba-server <2>
+stringData:
+  username: <username> <3>
+  password: <password> <4>
+--
+<1> Name of the Secret for the Samba server.
+<2> Namespace for the Secret for the Samba server.
+<3> Username for the Secret for the Samba server.
+<4> Password for the Secret for the Samba server.
+
+. Create a storage class by running the following command with the following example YAML file:
++
+[source,terminal]
+--
+$ oc create -f <sc_file_name>.yaml <1>
+--
+<1> Name of the storage class YAML file.
++
+[source,yaml]
+.Storage class example YAML file
+--
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: <sc_name> <1>
+provisioner: smb.csi.k8s.io
+parameters:
+  source: //<hostname>/<shares> <2>
+  csi.storage.k8s.io/provisioner-secret-name: smbcreds <3>
+  csi.storage.k8s.io/provisioner-secret-namespace: samba-server <4>
+  csi.storage.k8s.io/node-stage-secret-name: smbcreds <3>
+  csi.storage.k8s.io/node-stage-secret-namespace: samba-server <4>
+reclaimPolicy: Delete
+volumeBindingMode: Immediate
+mountOptions:
+  - dir_mode=0777
+  - file_mode=0777
+  - uid=1001
+  - gid=1001
+--
+<1> The name of the storage class.
+<2> The Samba server must be installed somewhere and reachable from the cluster with <`hostname>` being the hostname for the Samba server and `<shares>` the path the server is configured to have among the exported shares. 
+<3> Name of the Secret for the Samba server that was set in the previous step. If the `csi.storage.k8s.io/provisioner-secret` is provided, a a subdirectory is created with the PV name under `source`.
+<4> Namespace for the Secret for the Samba server that was set in the previous step.
+
+. Create a PVC:
+
+.. Create a PVC by running the following command with the following example YAML file:
```

Comment:
We need to highlight somewhere that provisioning volumes will will create a sub directory with the PV name under the "source" defined in the SC

Reply:
@gcharot This is mentioned in callout 3 under the SC YAML file in step 2. If this is not visible enough, I can pull it out into the text below the command for this step (this will make it more visible) and/or move it to step 3 when creating the PVC, if you think that would be helpful.

Reply:
Correct I missed it. We can keep it there but also add it to the Dynamic provisioning intro sentence?

Reply:
Done.